dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1535,17 +1553,48 @@ class CompileTest {
     };
   }
 
-  static Future<Map<String, dynamic>> getSizesFromIosApp(String appPath) async {
-    // Thin the binary to only contain one architecture.
-    final String xcodeBackend = path.join(flutterDirectory.path, 'packages', 'flutter_tools', 'bin', 'xcode_backend.sh');
-    await exec(xcodeBackend, <String>['thin'], environment: <String, String>{
-      'ARCHS': 'arm64',
-      'WRAPPER_NAME': path.basename(appPath),
-      'TARGET_BUILD_DIR': path.dirname(appPath),
-    });
+  static Future<Map<String, dynamic>> getSizesFromDarwinApp({
+    required String appPath,
+    required DeviceOperatingSystem operatingSystem,
+  }) async {
+    if(!operatingSystem.isDarwin) {
+      throw Exception('$operatingSystem is not a darwin os.');
+    }
+
+    late final String frameworkDirectory;
+    if (operatingSystem == DeviceOperatingSystem.ios) {
+      frameworkDirectory = path.join(
+        appPath,
+        'Frameworks',
+      );
+    } else if (operatingSystem == DeviceOperatingSystem.macos) {
+      frameworkDirectory = path.join(
+        appPath,
+        'Contents',
+        'Frameworks',
+      );
+    }
 
-    final File appFramework = File(path.join(appPath, 'Frameworks', 'App.framework', 'App'));
-    final File flutterFramework = File(path.join(appPath, 'Frameworks', 'Flutter.framework', 'Flutter'));
+    final File appFramework = File(path.join(
+      frameworkDirectory,
+      'App.framework',
+      'App',
+    ));
+
+    late final File flutterFramework;
+    if (operatingSystem == DeviceOperatingSystem.ios) {
```

Comment:
I know you copied this from me but you could have just one if and set `frameworkDirectory` and `flutterFramework` at the same time.

Reply:
With this way, we use one conditional tree for each assignment. You are suggesting that we use one tree for all assignments. I personally think that this way is more readable because declaration and assignment logic are contained to one region of the file. More importantly, that region is not mixed with other assignments. What do you think about that?

Reply:
I was thinking this, but I don't really have a strong preference.
```dart
    late final String frameworkDirectory;
    late final File flutterFramework;
    if (operatingSystem == DeviceOperatingSystem.ios) {
      frameworkDirectory = path.join(
        appPath,
        'Frameworks',
      );
      flutterFramework = File(path.join(
        frameworkDirectory,
        'Flutter.framework',
        'Flutter'
      ));
    } else if (operatingSystem == DeviceOperatingSystem.macos) {
      frameworkDirectory = path.join(
        appPath,
        'Contents',
        'Frameworks',
      );
      flutterFramework = File(path.join(
        frameworkDirectory,
        'FlutterMacOS.framework',
        'FlutterMacOS'
      ));
    }

    final File appFramework = File(path.join(
      frameworkDirectory,
      'App.framework',
      'App',
    ));
```

Reply:
It's not exactly what you are suggesting, but I think this captures your suggestion as well as @gspencergoog 
https://github.com/flutter/flutter/blob/000cd8bb89ee4abf9f7fc49163a76a7cca74712d/dev/devicelab/lib/tasks/perf_tests.dart#L1560-L1593